components/automate-deployment/pkg/server/server.go

```diff
@@ -437,20 +439,26 @@ func (s *server) configRenderer() (ConfigRenderer, error) {
 			return "", errors.Wrapf(err, "could not converge %s configuration to TOML", service.Name())
 		}
 
-		if usesPlatformScaffolding(service) {
+		pkgsMeta := make([]*product.PackageMetadata, 0, len(s.deployment.ExpectedServices))
```

Comment:
why do we need to gather all the metadatas every time for each service? Can we just grab `services.MetadataForPackage(service.Name())`? Or maybe `services.UsesPlatformScaffolding(service.Name())`?